### PR TITLE
Update ember-cli and fix failing tests

### PR DESCRIPTION
**Pull request by GavinJoyce**
In order to continue working on [MU Blueprints](https://github.com/ember-cli/ember-cli/issues/7530), we need to update the ember-cli version that we're using in ember.js.

The output of `ember new` has since been changed to emit new style tests. These changes result in a number of failing ember.js tests. For example, the `destroyApp` helper no longer exists on disk so the default generated test format output has changes for a bunch of `ember generate` blueprints.

I've updated the existing tests to expect the new default output and I've pinned `ember-cli` at [this SHA](https://github.com/ember-cli/ember-cli/commit/8f5a13a02d8ea5a04683565340611161c63bef89).

I've also [removed ember-cli config caching](https://github.com/ember-cli/ember-cli/pull/7739) which was also causing failing tests ember-source

### CODE REVIEW

**Comment by rwjblue** [comment]
> Are we happy to continue testing the (updated) default generator output as before and live with similar to the existing gaps in our coverage?

Yes.

File: node-tests/blueprints/component-test-test.js [hunk 1]
```diff
@@ -49,7 +49,8 @@ describe('Blueprint: component-test', function() {
       });
     });
 
-    describe('with usePods=true', function() {
+    //TODO: GJ: figure out what changed here
+    describe.skip('with usePods=true', function() {
```
Comment (rwjblue) [inline node-tests/blueprints/component-test-test.js]: Maybe one of these:

https://github.com/ember-cli/ember-cli/pull/7164
https://github.com/ember-cli/ember-cli/pull/7491

Comment (GavinJoyce) [reply node-tests/blueprints/component-test-test.js]: thanks, a git bisect points to https://github.com/ember-cli/ember-cli/commit/5d5801a086ae5789d771a7e4ae158392e9fdf6b0

Comment (rwjblue) [reply node-tests/blueprints/component-test-test.js]: My guess is due to caching 🤔

Comment (GavinJoyce) [reply node-tests/blueprints/component-test-test.js]: Yes, removing the caching fixes the issue

https://github.com/ember-cli/ember-cli/blob/252e1ccd1a99222397c8c9b293eb5e5f376070e1/lib/utilities/get-config.js#L18

Comment (GavinJoyce) [reply node-tests/blueprints/component-test-test.js]: oh, there are some details on this in the previous PR: https://github.com/ember-cli/ember-cli/pull/7491#issuecomment-356756278

Comment (GavinJoyce) [reply node-tests/blueprints/component-test-test.js]: and here: https://github.com/ember-cli/ember-cli/pull/7491#discussion_r162199039

I'll sync up with @patocallaghan in the morning and we'll try to remedy this

Comment (rwjblue) [reply node-tests/blueprints/component-test-test.js]: I think in general that the caching is fine, we just need an escape hatch to bypass and reset...

tldr; cache invalidation is hard

Comment (patocallaghan) [reply node-tests/blueprints/component-test-test.js]: @GavinJoyce I wasn't aware that the tests had been moved elsewhere, I just thought they'd been deleted.

I had a look there and I remember it being quite tricky to make those particular tests work with `get-config`. The config had already been cached at the module level and the `replaceFile` logic in the test happened too late. It wasn't easy to "reach in". In other tests that tested this sort of thing we used multiple fixture directories so it worked fine.

We're not actually using this feature ourselves yet so if it unblocks your MU work I'm fine for the changes to be reverted for now.

@rwjblue any thoughts? another option is to remove the cache but I believe the instrumentation lib will call `getConfig` a lot which will result in lots of `Yam` objects being created.

Comment (GavinJoyce) [reply node-tests/blueprints/component-test-test.js]: Two options:

 * [remove config caching](https://github.com/ember-cli/ember-cli/pull/7739)
 * [revert "Allow instrumentation to be turned on using .ember-cl"](https://github.com/ember-cli/ember-cli/pull/7738)

**Comment by GavinJoyce** [comment]
@rwjblue this is now 🍏 and ready for review

File: blueprints/initializer-test/qunit-files/tests/unit/initializers/__name__-test.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { run } from '@ember/runloop';
 
 import { initialize } from '<%= dasherizedModulePrefix %>/initializers/<%= dasherizedModuleName %>';
 import { module, test } from 'qunit';
-<% if (destroyAppExists) { %>import destroyApp from '../../helpers/destroy-app';<% } else { %>import { run } from '@ember/runloop'; <% } %>
```
Comment (GavinJoyce) [inline blueprints/initializer-test/qunit-files/tests/unit/initializers/__name__-test.js]: `run` is already imported above

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you @GavinJoyce!
